src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -446,6 +446,69 @@ internal static (string processFilePath, string commandLineArguments) GetServerP
             return (processFilePath, commandLineArgs);
         }
 
+        /// <summary>
+        /// Creates an environment block for Windows CreateProcess API.
+        /// </summary>
+        /// <param name="environmentVariables">Dictionary of environment variables to include</param>
+        /// <returns>Pointer to environment block that must be freed with <see cref="Marshal.FreeHGlobal"/></returns>
+        private static IntPtr CreateEnvironmentBlock(Dictionary<string, string> environmentVariables)
+        {
+            if (environmentVariables.Count == 0)
+            {
+                return IntPtr.Zero;
+            }
+
+            // Build the environment block as a single string
+            var envBlock = new StringBuilder();
+            foreach (var kvp in environmentVariables.OrderBy(kvp => kvp.Key, StringComparer.OrdinalIgnoreCase))
```

Comment:
Please add a comment indicating why ordering using OrdinalIgnoreCase is needed when creating the environment block